crates/optimism/payload/src/builder.rs

```diff
@@ -620,7 +625,15 @@ where
         let base_fee = builder.evm_mut().block().basefee;
 
         while let Some(tx) = best_txs.next(()) {
+            let interop = tx.interop();
             let tx = tx.into_consensus();
+            // Check that cross chain tx is validated by supervisor for this block
+            if let Some(interop) = interop {
+                if !interop.is_valid(self.config.attributes.timestamp()) {
+                    best_txs.mark_invalid(tx.signer(), tx.nonce());
```

Comment:
this line needs to be removed! otherwise the txns have no chance of ever becoming valid

Reply:
Okay, makes sense, because we will remove them on the next block update anyway

Reply:
But i think mark_invalid would work too, because we validate txs upon receiving for 1h, so we won't have a case when transaction that wasn't validated ends up in pending pool

Reply:
right, so we can guarantee this `mark_invalid` only effects the virtualisation `BestTransactions` over the pool, i.e. no message is sent to tx pool to change the status of the tx to invalid or similar

Reply:
the mark_invalid only affects the `BestTransactions` iterator, see:

https://github.com/paradigmxyz/reth/pull/15348#discussion_r2018335298